src/Internal.AspNetCore.BuildTools.Tasks/GetDotNetHost.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if NETSTANDARD1_6
+
+using System;
+using System.Runtime.InteropServices;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.BuildTools
+{
+    /// <summary>
+    /// Uses the current runtime information to find the "dotnet.exe" path.
+    /// This requires that MSBuild itself execute on top of Microsoft.NETCore.App.
+    /// </summary>
+    public class GetDotNetHost : Task
```

Comment:
Can we use the code from https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.Cli.Utils/Muxer.cs as is (and add a back link)? We could get away with not updating the TFM. 
I'm guessing we wouldn't be able to use the package directly without some packaging shenanigans.